core/bin/external_node/src/config/mod.rs

```diff
@@ -874,6 +1205,64 @@ impl ExternalNodeConfig<()> {
         })
     }
 
+    pub fn from_files(
```

Comment:
So here's how embedding configs in a single file could work:

```rust
// Could make sense to eventually add to `zksync_protobuf::serde`
#[derive(Debug)]
pub struct SerdeRepr<T: ProtoRepr>(pub T::Type);

impl<'de, T: ProtoRepr> Deserialize<'de> for SerdeRepr<T> {
    fn deserialize<D: Deserializer<'de>>(deserializer: D) -> Result<Self, D::Error> {
        let repr: T = deserialize_proto(deserializer)?;
        repr.read().map(Self).map_err(D::Error::custom)
    }
}

#[derive(Debug, Deserialize)]
struct WrappingConfig {
    external_node: SerdeRepr<proto::en::ExternalNode>,
    consensus: Option<SerdeRepr<proto::consensus::Config>>,
    #[serde(flatten)]
    general: SerdeRepr<proto::general::GeneralConfig>,
}

// usage
let yaml = std::fs::read_to_string(&path).with_context(|| path.display().to_string())?;
let config: WrappingConfig = serde_yaml::from_str(&yaml)?;
```

Reply:
I generally like it, but we don't need to implement it.